library/src/scala/Tuple.scala

```diff
@@ -200,44 +260,55 @@ object Tuple {
    */
   type IsMappedBy[F[_]] = [X <: Tuple] =>> X =:= Map[InverseMap[X, F], F]
 
-  /** Type of the reversed tuple */
-  @experimental
+  /** A tuple with the elements of tuple `X` in reversed order */
   type Reverse[X <: Tuple] = ReverseOnto[X, EmptyTuple]
 
-  /** Prepends all elements of a tuple in reverse order onto the other tuple */
-  @experimental
-  type ReverseOnto[From <: Tuple, +To <: Tuple] <: Tuple = From match
-    case x *: xs => ReverseOnto[xs, x *: To]
-    case EmptyTuple => To
-
-  /** Transforms a tuple `(T1, ..., Tn)` into `(T1, ..., Ti)`. */
-  type Take[T <: Tuple, N <: Int] <: Tuple = N match {
-    case 0 => EmptyTuple
-    case S[n1] => T match {
-      case EmptyTuple => EmptyTuple
-      case x *: xs => x *: Take[xs, n1]
-    }
-  }
-
-  /** Transforms a tuple `(T1, ..., Tn)` into `(Ti+1, ..., Tn)`. */
-  type Drop[T <: Tuple, N <: Int] <: Tuple = N match {
-    case 0 => T
-    case S[n1] => T match {
-      case EmptyTuple => EmptyTuple
-      case x *: xs => Drop[xs, n1]
-    }
-  }
-
-  /** Splits a tuple (T1, ..., Tn) into a pair of two tuples `(T1, ..., Ti)` and
-   * `(Ti+1, ..., Tn)`.
-   */
-  type Split[T <: Tuple, N <: Int] = (Take[T, N], Drop[T, N])
+  /** A tuple with the elements of tuple `X` in reversed order added in front of `Acc` */
+  type ReverseOnto[X <: Tuple, Acc <: Tuple] <: Tuple = X match
+    case x *: xs => ReverseOnto[xs, x *: Acc]
+    case EmptyTuple => Acc
 
   /** Given a tuple `(T1, ..., Tn)`, returns a union of its
    *  member types: `T1 | ... | Tn`. Returns `Nothing` if the tuple is empty.
    */
   type Union[T <: Tuple] = Fold[T, Nothing, [x, y] =>> x | y]
 
+  /** A type level Boolean indicating whether the tuple `X` conforms
+   *  to the tuple `Y`. This means:
+   *   - the two tuples have the same number of elements
+   *   - for corresponding elements `x` in `X` and `y` in `Y`, `x` matches `y`.
+   *  @pre  The elements of `X` are assumed to be singleton types
+   */
+  type Conforms[X <: Tuple, Y <: Tuple] <: Boolean = Y match
+    case EmptyTuple =>
+      X match
+        case EmptyTuple => true
+        case _ => false
+    case y *: ys =>
+      X match
+        case `y` *: xs => Conforms[xs, ys]
+        case _ => false
+
+  /** A type level Boolean indicating whether the tuple `X` has an element
+   *  that matches `Y`.
+   *  @pre  The elements of `X` are assumed to be singleton types
+   */
+  type Contains[X <: Tuple, Y] <: Boolean = X match
+    case Y *: _ => true
+    case x *: xs => Contains[xs, Y]
+    case EmptyTuple => false
+
+  /** A type level Boolean indicating whether the type `Y` contains
+   *  none of the elements of `X`.
+   *  @pre  The elements of `X` and `Y` are assumed to be singleton types
+   */
+  type Disjoint[X <: Tuple, Y <: Tuple] <: Boolean = X match
+    case x *: xs =>
+      Contains[Y, x] match
+        case true => false
+        case false => Disjoint[xs, Y]
```

Comment:
Why not `![Contains[Y, x]] && Disjoint[xs, Y]` ?

Reply:
Does not compile. Prefix operators are not allowed for types. I think we are lacking a type-level `!` currently.

Reply:
Yes it requires the brackets around the negated boolean type (and an `import compiletime.ops.boolean.*`)

Reply:
@odersky should I revive the prefix types SIP? https://github.com/scala/improvement-proposals/pull/35

Reply:
I don't really have an opinion. It's not strictly necessary, just a syntactic convenience.